crates/cadenza-eval/src/value.rs

```diff
@@ -5,6 +5,58 @@
 use crate::{diagnostic::Result, interner::InternedString};
 use std::fmt;
 
+/// A runtime type in the Cadenza evaluator.
+///
+/// Types are first-class values that can be inspected and operated on at runtime.
+/// This allows for type-level programming and better error messages.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum Type {
```

Comment:
Additionally we will need:
* a record type with field names and types
* A tuple type with a list of types
* An enum type with a list of variant names and types